Assets/BowAndArrowImproved/Scripts/TeleportManager.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TeleportManager : MonoBehaviour
+{
+    [SerializeField] private GameObject playerContainer;
+    [SerializeField] private List<TeleportAnchor> anchorsList;
+
+    private Coroutine _teleportCoroutine;
+    
+    private void OnEnable()
+    {
+        _teleportCoroutine = StartCoroutine(TeleportCoroutine());
+    }
+
+    private void OnDisable()
+    {
+        if (_teleportCoroutine != null)
```

Comment:
this is not smart at all, what if you stop the in the middle of the coroutine execution? half of the anchors will be enabled and half will be disabled!

disable all anchors here after stopping the coroutine to make sure no anchor is left enabled